sql/src/test/java/org/apache/druid/sql/calcite/CalciteWindowQueryTest.java

```diff
@@ -73,6 +73,13 @@ public class CalciteWindowQueryTest extends BaseCalciteQueryTest
       QueryContexts.ENABLE_DEBUG, true
   );
 
+  private static final Map<String, Object> DEFAULT_QUERY_CONTEXT_WITH_SUBQUERY_BYTES =
+      ImmutableMap.<String, Object>builder()
+                  .putAll(DEFAULT_QUERY_CONTEXT)
+                  .put(QueryContexts.MAX_SUBQUERY_BYTES_KEY, "100000")
+                  .put(QueryContexts.MAX_SUBQUERY_ROWS_KEY, "0")
```

Comment:
why the need to set `MAX_SUBQUERY_ROWS_KEY` to `0` ?

we have default as 100k and `@Min(1)` in [ServerConfig defaults](https://github.com/apache/druid/blob/71725b41b54d13e1d4bd6c8b4f9b2972790d3dea/server/src/main/java/org/apache/druid/server/initialization/ServerConfig.java#L138-L140) 
so without setting it like this its not even possible to set it to `0` 
interestingly the way this value is read from the context could even read negative values...
another interesting thing is that only `<0` is considered as unset  in [ClientQuerySegmentWrangler](https://github.com/apache/druid/blob/71725b41b54d13e1d4bd6c8b4f9b2972790d3dea/server/src/main/java/org/apache/druid/server/ClientQuerySegmentWalker.java#L669)

Reply:
>  so without setting it like this its not even possible to set it to 0

With maxSubqueryBytes set, the queries can default to row-based limiting. For example, here: https://github.com/apache/druid/pull/16800/files#diff-788b8a750f09134a406d02535f5d06b106ce89e3aece072902af3362b310ac83L3. Even though maxSubqueryBytes was set, the query passed even though window functions don't support byte-based limiting. 

> we have default as 100k and @Min(1) in [ServerConfig defaults](https://github.com/apache/druid/blob/71725b41b54d13e1d4bd6c8b4f9b2972790d3dea/server/src/main/java/org/apache/druid/server/initialization/ServerConfig.java#L138-L140)

Server config defaults are for the whole server, therefore it isn't a good practice to set it to something negative which will be interpreted as unlimited. For individual queries, however, we can bypass it as needed.

Reply:
I see now - There isn't a check in the query context that disallows the maxSubqueryBytes to be set to 0 using the query context. If it is negative, the `ClientQuerySegmentWalker` throws an exception, else it doesn't. So 0 in the query context accidentally does what it means, even though the value is disallowed on a cluster level. Given that it has been like this for a long time, should we keep it as is for this PR (I think negative -> unset is also weird, but any change to the semantics here might break queries of the existing clusters).